src/Umbraco.Core/Strings/DefaultShortStringHelper.cs

```diff
@@ -305,10 +305,10 @@ protected virtual string CleanString(string text, CleanStringType stringType, st
             return text;
         }
 
-        private static string RemoveSurrogatePairs(string text)
+        private string RemoveSurrogatePairs(string text)
         {
-            var input = text.ToCharArray();
-            var output = new char[input.Length];
+            var input = text.AsSpan();
+            Span<char> output = input.Length <= 1024 ? stackalloc char[input.Length] : new char[text.Length];
```

Comment:
Perhaps use https://learn.microsoft.com/en-us/dotnet/api/microsoft.toolkit.highperformance.buffers.spanowner-1?view=win-comm-toolkit-dotnet-7.0 SpanOwner or MemoryOwner to rent a buffer. https://learn.microsoft.com/en-us/dotnet/api/microsoft.toolkit.highperformance.buffers.memoryowner-1?view=win-comm-toolkit-dotnet-7.0. Ideally this method would return Span<char> or Memory<char>. The less times a string gets created, the better.